src/odemis/driver/xt_client.py

```diff
@@ -2279,6 +2286,51 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 # know the user that the move is not complete.
                 if future._must_stop.is_set():
                     raise CancelledError()
+
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
+                # TODO: base the timeout on the stage movement time estimation (est. *10)
+                timeout = time.time() + 10  # s
+                not_updated = True
+                wait_duration = 20e-03  # s
+
+                # Check for presence of rx/t or rz/r keys in pos
+                # In FM mode, the stage will maintain its rotation and tilt, if it is requested to change
+                # it is a different imaging mode
+                # TODO: extend the behavior to other imaging modes
+                if 'r' in pos or 't' in pos:
+                    logging.debug("Position requested to move includes rx or rz. Reported position accuracy is not"
+                                  "checked.")
+                    not_updated = False
+
+                # Find target position in absolute
+                if not rel:
+                    target_pos = pos
+                else:
+                    target_pos = {ax: val + pos[ax] for ax, val in orig_pos_raw.items()}
+
+                while not_updated:
+                    time.sleep(wait_duration)
+                    if time.time() > timeout:
+                        logging.warning("Stage position after move + %s s is still same as original pos: %s. "
+                                        "Giving up waiting", 10, orig_pos_raw)
+                        break
+
+                    # Every axis requested to move should have moved (compared to the position before starting)
+                    current_pos_raw = self.parent.get_stage_position()
+                    if not isNearPosition(orig_pos_raw, current_pos_raw, axes=set(orig_pos_raw.keys())):
```

Comment:
can you add a comment to explain why you check all the keys here, while in the next step you only check x and y?

Reply:
And is it not necessary to define a atol_linear here? The default is 100µm, so if your move is less than 100µm, you will never go inside this if statement.